wgkex/client/app.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+import hashlib
+import re
+
+from netlink import find_stale_wireguard_clients, link_handler, WireGuardClient
+
+from dataclasses import dataclass
+from salt.utils.network import mac2eui64
+from textwrap import wrap
+from typing import Dict, List
+
+# we receive stuff from wgkex-broker
+def generate_lladdr(public_key: str) -> str:
+    m = hashlib.md5()
+
+    m.update(public_key.encode("ascii") + b"\n")
+    hashed_key = m.hexdigest()
+    hash_as_list = wrap(hashed_key, 2)
+    temp_mac = ":".join(["02"] + hash_as_list[:5])
+
+    lladdr = re.sub("\/\d+$", "/128", mac2eui64(mac=temp_mac, prefix="fe80::/10"))
+    return lladdr
+
+
+def generate_interface_names(peer: WireGuardClient) -> WireGuardClient:
+    peer.wg_interface = "wg-" + peer.domain
+    peer.vx_interface = "vx-" + peer.domain
+
+    return peer
+
+
+def cleanup_wireguard_clients(domain: str) -> bool:
```

Comment:
That result type is wrong.  `link_handler`  returns either an empty list or a `Dict` (specifically a `Dict[str, Dict]` it seems).

I suspect you meant to return a `list[Dict[str, Dict]]` ?